cst/plugins/eu.esdihumboldt.cst.functions.numeric/src/eu/esdihumboldt/cst/functions/numeric/sequentialid/SequentialID.java

```diff
@@ -86,18 +84,24 @@ protected Object evaluate(String transformationIdentifier, TransformationEngine
 				id = seqValue + 1;
 			}
 			else {
+				if (Integer.parseInt(startValue) == 0) {
```

Comment:
I think that the changes that were proposed in the ticket will not meet by these code changes.
If I understand the requirement correctly then Kate and Thorsten mentioned that the ID should not start with leading 0's, i.e. 0011, 0001, 0000, etc.
Moreover, with your existing changes that happens, i.e. `leading zeros are stripped in the data`. The only thing you need too modify/do 
- is in the `StartValue` text box where the user enters the string: user should not be able to start a value from 0. If a user does this then you need to show some kind of error message in the dialog box saying `a start value cannot start with 0`. 
- `ID example` label to not show any example that has start value starting with 0. (if you do the first steps then this will automatically be handled.) 
With the changes that you have done

Reply:
I don't agree. I have tested my implementation: if one put 0 as `startValue` then the transformation ends with an error and in `Details` one can see the exception message `"ERROR: the starting value for ID creation should be larger than 0"`.

However, maybe I can also implement your suggestions (in the bullet points) for a more robust fixup.

What do you think?